src/main/java/roomescape/dao/ReservationDAO.java

```diff
@@ -0,0 +1,80 @@
+package roomescape.dao;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+import roomescape.exception.NotFoundReservationException;
+import roomescape.domain.Reservation;
+import roomescape.domain.Time;
+
+import java.sql.PreparedStatement;
+import java.util.List;
+
+@Repository
+public class ReservationDAO {
+
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    public ReservationDAO(JdbcTemplate jdbcTemplate, TimeDAO timeDAO) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    private final RowMapper<Reservation> rowMapper = (resultSet, rowNum) -> {
+        Time time = new Time(
+                resultSet.getLong("time_id"),
+                resultSet.getString("time_value")
+        );
+        return new Reservation(
+                resultSet.getLong("reservation_id"),
+                resultSet.getString("name"),
+                resultSet.getString("date"),
+                time
+        );
+    };
+
+    public List<Reservation> findAllReservations() {
+        String sql = "SELECT r.id as reservation_id, r.name, r.date, t.id as time_id, t.time as time_value " +
+                        "FROM reservation as r " +
+                        "inner join time as t on r.time_id = t.id";
+        return jdbcTemplate.query(sql, rowMapper);
+    }
```

Comment:
```suggestion
    public List<Reservation> findAllReservations() {
        String sql = """
                         SELECT r.id as reservation_id, r.name, r.date, t.id as time_id, t.time as time_value
                         FROM reservation as r inner join time as t on r.time_id = t.id
                         """;
        return jdbcTemplate.query(sql, rowMapper);
    }
```
- 이런식으로 텍스트 블럭을 사용하면 +연산자 없이 더 깔끔히 선언할 수 있을 것 같아요.
- [Java-Multiline-String](https://dejavuhyo.github.io/posts/java-multiline-string/)

Reply:
넵!!